docs/file-conventions/routes.md

```diff
@@ -250,7 +250,7 @@ Wrapping a route segment in parentheses will make the segment optional.
 | `/categories`              | `app/routes/($lang).categories.tsx` |
 | `/en/categories`           | `app/routes/($lang).categories.tsx` |
 | `/fr/categories`           | `app/routes/($lang).categories.tsx` |
-| `/american-flag-speedo`    | `app/routes/($lang)._index.tsx`     |
+| `/american-flag-speedo`    | `app/routes/($lang).$productId.tsx` |
```

Comment:
Actually, this is correct as-is. Optional dynamic segments match eagerly so in this case when you olly have one URL segment it will be consumed by `$lang` and you'll see the `routes/($lang)._index.tsx` route